src/lib.rs

```diff
@@ -316,12 +321,23 @@ impl Message {
             .finalize()[12..]
             != self.address
         {
+            #[cfg(feature = "ethers")]
+            if self.verify_eip1271(sig).await? {
+                return Ok(());
+            }
             Err(VerificationError::Signer)
         } else {
-            Ok(pk.to_bytes().into_iter().collect())
+            Ok(())
         }
     }
 
+    #[cfg(feature = "ethers")]
+    pub async fn verify_eip1271(&self, sig: &[u8]) -> Result<bool, VerificationError> {
```

Comment:
Could you add some doc comments?

Reply:
Yes, coming in a follow-up commit/PR with the config object